subprojects/language-groovy/src/main/java/org/gradle/api/tasks/javadoc/Groovydoc.java

```diff
@@ -344,18 +350,80 @@ public void setOverviewText(@Nullable TextResource overviewText) {
     }
 
     /**
-     * Returns whether to include all classes and members (i.e. including private ones).
+     * Returns whether to include classes with private access and above.
+     *
+     * @deprecated Equivalent to calling {@link #getAccess()}{@code .get()} and checking for equivalence with {@link GroovydocAccess#PRIVATE}
      */
-    @Input
+    @Internal
+    @Deprecated
     public boolean isIncludePrivate() {
-        return includePrivate;
+        DeprecationLogger.deprecateMethod(Groovydoc.class, "isIncludePrivate()")
+            .replaceWith("getAccess()")
+            .willBeRemovedInGradle8()
+            .withUpgradeGuideSection(7, "groovydoc_option_improvements")
+            .nagUser();
+        return getAccess().get() == GroovydocAccess.PRIVATE;
     }
 
     /**
-     * Sets whether to include all classes and members (i.e. including private ones) if set to true.
+     * Sets whether to include classes and members with private access and above.
+     *
+     * @deprecated Equivalent to calling {@link #setAccess(String)} with {@code "private"}
```

Comment:
setAccess(String) no longer exists

Reply:
@big-guy So `setAccess` was replaced with: 
```
getAccess().set(includePrivate ? GroovydocAccess.PRIVATE : GroovydocAccess.PUBLIC);
```

I assume the new strategy is strictly better, but it's awkward because all the other (14) properties in the file are still `setXyz`.  Not to mention it makes the deprecation comment a bit verbose.

It's definitely one of those annoying decisions: use better strategy on one new property, or be consistent with the rest of the file, or blow the scope up and refactor all the setters. I have no preference.

Reply:
Our preference currently is to break consistency, we've done this with many other classes before. I'll handle the fix.